app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt

```diff
@@ -1589,8 +1587,15 @@ class ConversationActivityV2 : PassphraseRequiredActionBarActivity(), InputBarDe
         val text = getMessageBody()
         val userPublicKey = textSecurePreferences.getLocalNumber()
         val isNoteToSelf = (recipient.isContactRecipient && recipient.address.toString() == userPublicKey)
-        if (text.contains(seed) && !isNoteToSelf && !hasPermissionToSendSeed) {
-            start<RecoveryPasswordActivity>()
+        if (seed in text && !isNoteToSelf && !hasPermissionToSendSeed) {
+            showSessionDialog {
+                title(R.string.dialog_send_seed_title)
+                text(R.string.dialog_send_seed_explanation)
+                button(R.string.dialog_send_seed_send_button_title) { sendTextOnlyMessage(true) }
```

Comment:
Would it be better to have a mutable copy of the argument which we can flip to true rather than calling ourselves, and then we just return null on cancel?

Reply:
Sorry, this one was mostly a revert so I merged w/one

IDK if I understand, but it could make sense for `showDialog` or some variant to be a suspending function like `Snackbar` is... 🤔 so you call `val result = showDialog(...)`

but right now it's not true recursion, as `sendTextOnlyMessage(true)` is called in a listener as a response to a click.